sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/PostProcessor.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue;
+
+import com.azure.storage.common.Utility;
+import com.azure.storage.queue.models.StorageErrorException;
+import com.azure.storage.queue.models.StorageException;
+import reactor.core.publisher.Mono;
+
+final class PostProcessor {
+    static <T> Mono<T> postProcessResponse(Mono<T> response) {
+        return Utility.postProcessResponse(response, (errorResponse) ->
+            errorResponse.onErrorResume(StorageErrorException.class, resume ->
+                resume.response()
+                    .bodyAsString()
+                    .switchIfEmpty(Mono.just(""))
+                    .flatMap(body -> Mono.error(new StorageException(resume, body)))
+            ));
+    }
+
+    private PostProcessor() {
+    }
+}
```

Comment:
it would be interesting to have just one PostProcessor.java file in storage/common to support queue and file..   Looks like the difference is just the models used for the Exception.

Reply:
That would result in a cyclical dependency as Blobs, Files, and Queues have a dependency on Common and that would force Common to take a dependency on them.

Reply:
I see.  that would require to split commons package in 2 right? `utils` for reusable dependency 
 between Blobs for Files and Queues and `shared` as a generic class with Blobs, Files and Queues as dependency XD..  First one is used by each lib, while second one is used by users...
But, just an idea, it might be complex to refactor current impl for this

The only win will be to avoid having 3 packages for each shared class like the PostProcessor..  don't know if it is the only class like that or if there would be more like it

Reply:
For now let's leave it as is and extend in the future